src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.SslCtx.cs

```diff
@@ -69,7 +83,107 @@ protected override bool ReleaseHandle()
         {
             Interop.Ssl.SslCtxDestroy(handle);
             SetHandle(IntPtr.Zero);
+            if (_gch.IsAllocated)
+            {
+                //Interop.Ssl.SslCtxSetData(this, (IntPtr)_gch);
+                _gch.Free();
+            }
+
+            if (_sslSessions != null)
+            {
+                lock (_sslSessions)
+                {
+                    foreach (string name in _sslSessions.Keys)
+                    {
+                        _sslSessions.Remove(name, out IntPtr session);
+                        Interop.Ssl.SessionFree(session);
+                    }
```

Comment:
Isn't this just

```suggestion
                    foreach (IntPtr session in _sslSessions.Values)
                    {
                        Interop.Ssl.SessionFree(session);
                    }
   
                    _sslSessions.Clear();
```

that way it doesn't do an N*log(N) re-fetch loop.

Reply:
yes. updated.